sound/soc/sof/intel/hda-loader.c

```diff
@@ -337,8 +337,11 @@ int hda_dsp_cl_boot_firmware_iccmax(struct snd_sof_dev *sdev)
 
 	stripped_firmware.size = sdev->basefw.fw->size - sdev->basefw.payload_offset;
 
-	/* prepare capture stream for ICCMAX */
-	iccmax_stream = hda_cl_stream_prepare(sdev, HDA_CL_STREAM_FORMAT, stripped_firmware.size,
```

Comment:
The `if (sdev->basefw.fw->size <= sdev->basefw.payload_offset)` still remained, it can also dropped as it lost it's relevance (not that it had any anyways).